README.md

```diff
@@ -24,247 +21,310 @@
+### Node.js installation
 
-You can specify your choice using the `BACKEND` variable while running the script.
+Before running the installation steps, make sure that Node.js v18+ and npm are
+installed in your development environment.
 
-Install React along with Node.js backend.
-```
-BACKEND=nodejs ./setup.sh
-```
-
-Install React along with Python/Flask backend.
-```
-BACKEND=python source ./setup.sh
-```
+You can quickly install the required packages using the `package.json` file. 
+1. Navigate to the app directory, `server-js` (i.e. where `package.json` is
+   located).
+2. Run `npm install`. This will install all the required packages mentioned in
+   `package.json`.
 
-Install React along with both(Flask, and Node.js) backends.
-```
-BACKEND=all source ./setup.sh
-```
+### Python installation
 
-### Install manually(Linux/macOS/Windows)
+Before running the installation steps, make sure that Python 3.9+ is installed
+in your development environment.
 
-#### nvm(Node Version Manager) installation
+#### Create a virtual environment
 
 ##### Linux/macOS
-```
-curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.40.1/install.sh | bash
 
-export NVM_DIR="$HOME/.nvm"
-[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
-[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
 ```
-##### Windows
-Visit official [npm docs](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm#windows-node-version-managers) for installation instructions.
+python -m venv venv
+source venv/bin/activate
+```
 
-#### Node.js installation
-##### Linux/macOS/Windows
+##### Windows
 
-Install version **22.6.0** of node.
 ```
-nvm install 22.6.0
+python -m venv venv
+.\venv\Scripts\activate
 ```
-##### Install the package dependencies for Node.js
-You can quickly install the required packages using the `package.json` file. 
-1. Navigate to the app directory, `server-js` (i.e. where `package.json` is located).
-2. Run `npm install`. This will install all the required packages mentioned in `package.json`.
 
-#### Flask installation
+#### Install the required Python packages
 
-##### Create a new virtual environment.
-###### Linux/macOS
-   ```
-   python -m venv venv
-   source .venv/bin/activate
-   ```
-
-###### Windows
-   ```
-   python -m venv venv
-   .\venv\Scripts\activate
-   ```
+##### Linux/macOS/Windows
 
-##### Install the required Python packages.
-###### Linux/macOS/Windows
 ```
 pip install -r requirements.txt
 ```
 
-#### Go installation
+### Go installation
 
 Check if Go is already installed on your system.
+
 ```
 go version
 ```
-If Go is not installed, follow the instructions for your operating system from the [official Go installation guide](https://go.dev/doc/install).
+
+If Go is not installed, follow the instructions for your operating system from
+the [Go installation guide](https://go.dev/doc/install).
 
 ## Run the app
```

Comment:
The sub-headings below this one are hard to distinguish - I think that's adding to my confusion about which is required / optional.  e.g. when reading through Node, I tried to skim down to "Run the app" but there are three "run the app" sections the follow.

Could they go in separate setup files? Or maybe [collapsible sections](https://gist.github.com/pierrejoubert73/902cc94d79424356a8d20be2b382e1ab)?

Reply:
I added a comment that hopefully clarifies things...